src/local_scheduler/local_scheduler.cc

```diff
@@ -414,55 +503,42 @@ LocalSchedulerState *LocalSchedulerState_init(
   return state;
 }
 
-void update_dynamic_resources(LocalSchedulerState *state,
-                              TaskSpec *spec,
-                              bool return_resources) {
-  for (int i = 0; i < ResourceIndex_MAX; ++i) {
-    double resource = TaskSpec_get_required_resource(spec, i);
-    if (!return_resources) {
-      /* If we are not returning resources, we are leasing them, so we want to
-       * subtract the resource quantities from our accounting. */
-      resource *= -1;
-    }
+/**
+ * Check if a certain quantity of dynamic resources are available.
+ *
+ * @param state The state of the local scheduler.
+ * @param num_cpus Check if this many CPUs are available.
+ * @param num_gpus Check if this many GPUs are available.
+ * @return True if there are enough CPUs and GPUs and false otherwise.
+ */
+bool check_dynamic_resources(LocalSchedulerState *state,
+                             int64_t num_cpus,
+                             int64_t num_gpus) {
+  if (state->dynamic_resources[ResourceIndex_CPU] < num_cpus) {
+    return false;
+  }
+  if (state->dynamic_resources[ResourceIndex_GPU] < num_gpus) {
+    return false;
+  }
+  return true;
+}
 
-    bool oversubscribed =
-        (!return_resources && state->dynamic_resources[i] < 0);
-    /* Add or subtract the task's resources from our count. */
-    state->dynamic_resources[i] += resource;
-
-    if ((!return_resources && state->dynamic_resources[i] < 0) &&
-        !oversubscribed) {
-      /* Log a warning if we are using more resources than we have been
-       * allocated, and we weren't already oversubscribed. */
-      LOG_WARN("local_scheduler dynamic resources dropped to %8.4f\t%8.4f\n",
-               state->dynamic_resources[0], state->dynamic_resources[1]);
+bool sufficient_resources_for_task(LocalSchedulerState *state, TaskSpec *spec) {
+  bool sufficient_resources = true;
+  for (int i = 0; i < ResourceIndex_MAX; i++) {
+    if (TaskSpec_get_required_resource(spec, i) > state->dynamic_resources[i]) {
+      /* There are not enough resources for this task. */
+      sufficient_resources = false;
```

Comment:
Yeah, I initially wrote it that way, and I'm fine with either, but I've been trying to err on the side of having a single return statement per function.